bf.c

```diff
@@ -0,0 +1,107 @@
+#include <err.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#define CELLS_LEN 30000
+
+void interp(char *prog, char *prog_end, char *cells, char *cells_end) {
+    char *instr = prog;
+    char *cell = cells;
+    while (instr < prog_end) {
+        switch (*instr) {
+            case '>': {
+                if (cell++ == cells_end)
+                    errx(1, "out of memory");
+                instr++;
+                break;
+            }
+            case '<': {
+                if (cell > cells)
+                    cell--;
+                instr++;
+                break;
+            }
+            case '+': {
+                (*cell)++;
+                instr++;
+                break;
+            }
+            case '-': {
+                (*cell)--;
+                instr++;
+                break;
+            }
+            case '.': {
+                putchar(*cell);
+                instr++;
+                break;
+            }
+            case ',': {
+                if (read(STDIN_FILENO, cell, 1) == -1)
+                    err(1, "(stdin)");
+                instr++;
+                break;
+            }
+            case '[': {
+                if (*cell == 0) {
+                    int count = 0;
+                    while (true) {
+                        instr++;
+                        if (*instr == ']') {
+                            if (count == 0)
+                                break;
+                            count--;
+                        } else if (*instr == '[')
+                            count++;
+                    }
+                }
+                instr++;
+                break;
+            }
+            case ']': {
+                if (*cell != 0) {
+                    int count = 0;
+                    while (true) {
+                        instr--;
+                        if (*instr == '[') {
+                            if (count == 0)
+                                break;
+                            count--;
+                        } else if (*instr == ']')
+                            count++;
+                    }
+                }
+                instr++;
+                break;
+            }
+            default: {
+                instr++;
+                break;
+            }
+        }
+    }
+}
+
+int main(int argc, char *argv[]) {
+    if (argc < 2)
+        errx(1, "<filename>");
+
+    int fd = open(argv[1], O_RDONLY);
+    struct stat sb;
+    if (fstat(fd, &sb) != 0)
+        err(1, "%s", argv[1]);
+    char *prog = malloc(sb.st_size);
```

Comment:
mmap feels more appropriate than malloc here, maybe? Then the copying stage disappears.

Reply:
`mmap` feels like overkill to me (and, I must admit, I'm not super familiar with `mmap`'s edge cases; I can't see the use in me wasting my limited brain power on it for this little interpreter).

Reply:
OK, as you wish.

[mmap is a wonderful and underappreciated tool -- you should try it] :)